ksigner/src/jvmTest/kotlin/com/novoda/ksigner/hash/EncodeRequestTest.kt

```diff
@@ -0,0 +1,27 @@
+package com.novoda.ksigner.hash
+
+import org.junit.Test
+import kotlin.test.assertEquals
+
+class EncodeRequestTest {
+    private val canonicalRequest = "PUT\n" +
+            "/-/vaults/examplevault\n" +
+            "\n" +
+            "host:glacier.us-east-1.amazonaws.com\n" +
+            "x-amz-date:20120525T002453Z\n" +
+            "x-amz-glacier-version:2012-06-01\n" +
+            "\n" +
+            "host;x-amz-date;x-amz-glacier-version\n" +
+            "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
+
+
+    @ExperimentalUnsignedTypes
+    @Test
+    fun encodesRawCanonicalRequestToLowercaseHex() {
```

Comment:
This test has been ported from the [library repo](https://github.com/lucasweb78/aws-v4-signer-java/blob/master/src/test/java/uk/co/lucasweb/aws/v4/signer/Sha256Test.java) to make sure the KEthereum implementation of SHA-256 works as expected.